optimum/quanto/nn/qmodule.py

```diff
@@ -221,11 +222,17 @@ def qweight(self):
         if isinstance(self.weight, QTensor):
             # Frozen QModule
             return self.weight
-        # Quantize dynamically the weights per-axis
+
+        if self.weight_qtype == qfloat8_e4m3fn and self.activation_qtype is None:
+            # Marlin FP8 kernel only supports per-tensor fp8 quantization.
+            axis = None
+        else:
+            axis = 0
```

Comment:
Having this controlflow here is quite ugly.

Reply:
I don't understand why we have this restriction while the kernel checks for a scalar scale.

Reply:
Here is the kernel code:

```c++
  TORCH_CHECK(b_rank == 2, "b_scales rank = ", b_rank, " is not 2");
  TORCH_CHECK(b_scales.size(1) == size_n, "b_scales dim 1 = ", b_scales.size(1),
              " is not size_n = ", size_n);
  // Channelwise only for FP8
  TORCH_CHECK(b_scales.size(0) == 1)
  num_groups = b_scales.size(0);
```

So clearly NOT a scalar (this is why in the unit test you had to repeat your scalar scale).  I suspect the reason why it only works with a scalar scale (same value for all output features) is because of some kind of inter-leaving that is missing.